src/vivarium_gates_nutrition_optimization/components/children.py

```diff
@@ -1,11 +1,144 @@
 from pathlib import Path
+from typing import Tuple
 
+import numpy as np
 import pandas as pd
 from vivarium.framework.engine import Builder
 from vivarium.framework.event import Event
 from vivarium_cluster_tools.utilities import mkdir
 
-from vivarium_gates_nutrition_optimization.constants import models
+from vivarium_gates_nutrition_optimization.constants import (
+    data_keys,
+    data_values,
+    models,
+)
+
+
+class NewChildren:
+    def __init__(self):
+        self.lbwsg = LBWSGDistribution()
+
+    @property
+    def name(self):
+        return "child_status"
+
+    @property
+    def sub_components(self):
+        return [self.lbwsg]
+
+    @property
+    def columns_created(self):
+        return ["sex_of_child", "birth_weight", "gestational_age"]
```

Comment:
I understand why you want to calculate the gestational age in this component rather than the Pregnancy component, but it doesn't seem necessary to save this column in the population state table. Couldn't the `__call__` function just send this information back to the Pregnancy component without that component then creating it as a column? And in that case, the `empty()` function wouldn't need to return it at all - i.e. just send the `"sex_of_child"` and `"birth_weight"` columns back.

It's not a big deal either way.

Reply:
agreed, done